matlab/.editorconfig

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This is a EditorConfig file: https://editorconfig.org/
+
+# See ../.editorconfig for inherited values
+
+[*.{c,cc,cpp,h,hh,hpp}]
+indent_size = 1
```

Comment:
Could we use the inherited `indent_size` of **2** for C++ files for consistency with the rest of the language bindings?

Reply:
I agree, but here as well, it doesn't seem to match existing files?

Reply:
Along the same lines as my other comment - I don't believe the indentation level is fully consistent across the C++ code base for MATLAB currently. However, it's probably a good idea for us to strive for this moving forward.

I would have expected this to be enforced via https://github.com/apache/arrow/pull/40588, but it's likely I am missing something here.

Reply:
Will do as well.